modules/apps/application-list/application-list-api/src/main/java/com/liferay/application/list/BasePanelApp.java

```diff
@@ -99,15 +99,21 @@ public int getNotificationsCount(User user) {
 			return 0;
 		}
 
+		Portlet portlet = getPortlet();
+
 		return _userNotificationEventLocalService.
 			getUserNotificationEventsCount(
-				user.getUserId(), _portlet.getPortletId(),
+				user.getUserId(), portlet.getPortletId(),
 				UserNotificationDeliveryConstants.TYPE_WEBSITE, true, false);
 	}
 
 	@Override
 	public Portlet getPortlet() {
-		return _portlet;
+		if (_portletLocalService != null) {
```

Comment:
@W2jx Why this is needed? All sub classes should implement this method rather that have a implementation in base method.

Reply:
I will fix them.